// gatsby-theme-gallery/gatsby-node.js
const fs = require("fs");
const path = require("path");
const mkdirp = require("mkdirp");

// Ensure that content directory exist at the site-level
exports.onPreBootstrap = ({ reporter, store }, themeOptions) => {
  const { program } = store.getState();
  const galleryPath = themeOptions.galleryPath || "content/gallery";
  const dir = path.join(program.directory, galleryPath);
  if (!fs.existsSync(dir)) {
    reporter.info(`Creating the "${dir}" directory`);
    mkdirp.sync(dir);
  }
};

// exports.sourceNodes = ({ actions }) => {
//   const { createTypes } = actions;

//   createTypes(
//     schema.buildObjectType({
//       name: "GalleryImage",
//       fields: {
//         image: { type: `File!`, extensions: { fileByRelativePath: {} } },
//       },
//       interfaces: ["Node"],
//     }),
//   );
// };

exports.createPages = ({ actions }, themeOptions) => {
  const { createPage } = actions;
  const basePath = themeOptions.basePath || "/";

  createPage({
    path: basePath,
    component: path.resolve(`${__dirname}/src/pages/index.tsx`),
  });
};

// const supportedExtensions = {
//   jpeg: true,
//   jpg: true,
//   png: true,
//   webp: true,
//   tif: true,
//   tiff: true,
// };

// exports.onCreateNode = ({ node, actions, getNode }, themeOptions) => {
//   console.log("TCL: exports.onCreateNode -> node", node);
// const { createNode } = actions;
// const galleryPath = themeOptions.galleryPath || "content/gallery";

// if (!supportedExtensions[node.extension]) {
//   return;
// }

// const fileNode = getNode(node.parent);
// const source = fileNode.sourceInstanceName;

// console.log("TCL: exports.onCreateNode -> node", node);
// if (source === galleryPath) {
// createNode({
//   ...fieldData,
//   id: createNodeId(`${node.id} >>> Project`),
//   parent: node.id,
//   children: [],
//   internal: {
//     type: "GalleryImage",
//   },
// });
// }
// };
